kernel/fs.c

```diff
@@ -461,6 +473,7 @@ dword_t sys_pwrite(fd_t f, addr_t buf_addr, dword_t size, off_t_ off) {
         return _ENOMEM;
     if (user_read(buf_addr, buf, size))
         return _EFAULT;
+    enter_blocking_area();
```

Comment:
I think this can use the macro? There aren't any gotos in the way like sys_pread has.